Mage/src/main/java/mage/game/permanent/token/DragonEggDragonToken.java

```diff
@@ -42,5 +42,9 @@ public DragonEggDragonToken copy() {
     @Override
     public void setExpansionSetCodeForImage(String code) {
         super.setExpansionSetCodeForImage(code);
+
+        if (getOriginalExpansionSetCode() != null && getOriginalExpansionSetCode().equals("IMA")) {
+            this.setTokenType(1);
```

Comment:
No needs in setTokenType(1) -- it's default index.

Reply:
It's default index but because there's another token named "Dragon" in IMA it still needs to specify `setTokenType(1)` in the code, and `setTokenType(2)` in the other Dragon's code. I just tested this and the image doesn't work without the code above. I'm also about to push a commit which fixes this issue with M19's `DragonEggDragonToken` which I discovered while testing.

Reply:
Hmm. This is a bad behaviour or a bug. Must be research and fix or added verify test for it. Added TODO in #5622